sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -38,7 +39,7 @@ public class ServiceBusProcessor : IAsyncDisposable
 
         private readonly SemaphoreSlim _messageHandlerSemaphore;
 
-        private readonly int _maxConcurrentCalls;
+        private readonly object _syncLock = new();
```

Comment:
Should we consider calling this something like `maxConcurrencySync` or similar to denote that context that we're using it in?